docker/README.md

```diff
@@ -0,0 +1,130 @@
+## Running `teosd` in a docker container
+A `teos` image can be built from the Dockerfile located in `docker`. You can create the image by running:
+
+	cd rust-teos
+	docker build -f docker/Dockerfile -t teos .
+	
+Then we can create a container by running:
+
+	docker run -it teos
+
+To feed `teos` custom config options, there are two ways to do so: 
+
+1) Environment variables:
+
+Create a container by running:
+
+	docker run -it -e <ENV_VARIABLES> teos
+	
+Notice that the ENV variables are optional, if unset the corresponding default setting is used. The following ENVs are available:
+
+```
+- API_BIND=<teos_api_hostname>
+- API_PORT=<teos_api_port>
+- RPC_BIND=<teos_rpc_hostname>
+- RPC_PORT=<teos_rpc_port>
+- BTC_NETWORK=<btc_network>
+- BTC_RPC_CONNECT=<btc_node_hostname>
+- BTC_RPC_PORT=<btc_node_port>
+- BTC_RPC_USER=<btc_rpc_username>
+- BTC_RPC_PASSWORD=<btc_rpc_password>
+- DEBUG=<debug_bool>
+- DEPS_DEBUG=<deps_debug_bool>
+- OVERWRITE_KEY=<overwrite_key_bool>
+- FORCE_UPDATE=<force_update_bool>
+```
+
+2) Config file:
+
+If you'd like to use a `teos.toml` file instead, one option is to edit the Dockerfile with this line right above the ENTRYPOINT:
+
+	COPY teos.toml /home/teos/.teos
```

Comment:
I didn't realize this before, but `docker cp` won't actually work here because it can only be used once the container is running, but we need to copy over this data `before` it's running and we hit the endpoint. 

Can you elaborate on what you mean by the second point?

Reply:
> I didn't realize this before, but docker cp won't actually work here because it can only be used once the container is running, but we need to copy over this data before it's running and we hit the endpoint.

The way I did it (but think it's tedious tbh) is spin up a dummy alpine container with the volume just to copy the data over to the volume with `docker cp`, and then reuse the volume with `teos` image.

> Can you elaborate on what you mean by the second point?

When a volume is attached, any image data inside the mount point disappears, I guess what docker does is just replace the directory from the image with the volume content. So the `COPY` command provides some data at build time, which if you run the container without a volume would persist as you would expect, but if a volume is attached, the mount directory would be identical to what's inside the volume, effectively ignoring the build-time `COPY` command.

Reply:
Oh got it, interesting. Well one option in that case would be to add a VOLUME command above the COPY command in the Dockerfile to create the volume before rather than after

Personally, I don't think we don't need to get super deep in these instructions into... what if a user does this, what will happen? It's nice that we have some instructions to give users an outline, but I suspect most people using this Dockerfile will have some knowledge of Docker and will know how to tweak it for their specific needs

Reply:
Fine, let's remove the config file instructions and maybe just mention that such a method exists and leave it up to the user how to do so.